### PR TITLE
chore: break up main quiz data file into separate categories 

### PR DESCRIPTION
**Pull request by kingpinzs**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

I separated the type of questions into their own files.

### CODE REVIEW

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Hi @kingpinzs ! 

I appreciate your hark work and eagerness to contribute to the Developer Quiz site but there are some categories missing. 

The Developer Quiz site is a companion site for the Learn to Code RPG game. All of the questions we are using here will also be used over there. It is important that we line up with the categories that will be used over there. Quincy and the team that is working on that project has decided on the following categories:

HTML,
CSS,
JavaScript,
Python,
Linux,
Agile,
Security,
General IT,
Computer Science,
SQL,
freeCodeCamp,
Git, and 
Quality Assurance

You will need to add in those missing category files. 

We want to make sure that these recent agile additions make it into the new agile file
https://github.com/freeCodeCamp/Developer_Quiz_Site/commit/023d408295199d7709ca85b4e44f63ed981cb7c5
https://github.com/freeCodeCamp/Developer_Quiz_Site/commit/cddbf91bb5a0de18b9091a00656f6a96a9292301

We want to make sure that these recent security additions make it into the new security file
https://github.com/freeCodeCamp/Developer_Quiz_Site/commit/f015f6c555ac73d8af41fcf6834b42c2a01e11bb

In the future, it is best to open up a github discussion or issue when you are interested in working on features and making large changes. That way we can discuss it and agree on the work to be done 👍 

Please review the requested changes.
Once those changes have been implemented and approved, then we can merge your PR into main. 👍

File: src/data/full-quiz.ts [hunk 1]
```diff
@@ -7152,6 +7156,114 @@ const fullQuiz = [
...
+    Question:
+      "In C++ what value does an array start counting at",
+    Answer: "0",
+    Distractor1: "1",
+    Distractor2: "null",
+    Distractor3: "nil",
+    Explanation:
+      "Arrays start at index 0 and NOT index 1",
+    Link: "https://www.freecodecamp.org/news/how-to-learn-the-c-programming-language/",
+  },
```
[the remark this answers is not in the document]
Comment (kingpinzs) [reply src/data/full-quiz.ts:7262]: Full-quiz data file is no longer needed so I removed it

File: src/components/QuizTemplate.tsx [hunk 2]
```diff
@@ -17,7 +31,10 @@ interface QuizProps {
 }
 
 const QuizTemplate: React.FC<QuizProps> = (QuizProps) => {
-  const [quiz, setQuiz] = useState(fullQuiz);
+  const [quiz, setQuiz] = useState([...cssQuiz, ...freecodecampQuiz, 
+                                    ...generalCSQuiz, ...gitQuiz, ...htmlQuiz, ...informationTechnologyQuiz, 
+                                    ...javascriptQuiz, ...linuxQuiz, ...pythonQuiz, ...sqlQuiz,
+                                    ...agileQuiz, ...qualityAssuranceQuiz, ...securityQuiz]);
```
Comment (jdwilkin4) [inline src/components/QuizTemplate.tsx:37]: Let's create a variable here for all of the quiz categories. That way we only need to update it in one spot and it lessens the chance for someone in the future forgetting to add all of the categories in the `setQuiz `

```js
const allCategories = [...cssQuiz, ...freecodecampQuiz, 
                                    ...generalCSQuiz, ...gitQuiz, ...htmlQuiz, ...informationTechnologyQuiz, 
                                    ...javascriptQuiz, ...linuxQuiz, ...pythonQuiz, ...sqlQuiz,
                                    ...agileQuiz, ...qualityAssuranceQuiz, ...securityQuiz]
const [quiz, setQuiz] = useState(allCategories)
```

File: src/components/QuizTemplate.tsx [hunk 3]
```diff
@@ -75,7 +92,10 @@ const QuizTemplate: React.FC<QuizProps> = (QuizProps) => {
   };
 
   const resetQuiz = () => {
-    setQuiz(fullQuiz);
+    setQuiz([...cssQuiz, ...freecodecampQuiz, 
+      ...generalCSQuiz, ...gitQuiz, ...htmlQuiz, ...informationTechnologyQuiz, 
+      ...javascriptQuiz, ...linuxQuiz, ...pythonQuiz, ...sqlQuiz,
+      ...agileQuiz, ...qualityAssuranceQuiz, ...securityQuiz]);
```
Comment (jdwilkin4) [inline src/components/QuizTemplate.tsx:98]: Replace this with the `allCategories` variable I created above.

```js
 setQuiz(allCategories);
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]

**Review by jdwilkin4 — APPROVED** [review]
When I was running your branch locally, it had crashed because the tests weren't updated for the new categories. I went ahead and pushed up a commit to fix that. Normally, I wouldn't push to other people's branches but Hacktoberfest is around the corner and it was just quicker to fix it so the issues can be updated. 

I am approving this PR. Thanks for your hard work on this. 👍 
Any other feature/chore proposals you are interesetd in, please up a new discussion in the GitHub discussions tab and we can discuss it there.

**Pull request merged**